### ISSUE TITLE
[Feature Request]: Cochran-Mantel-Haenszel & McNemar test

### ISSUE THREAD

**Issue by Sal-Ramos** [jasp-stats/jasp-issues#1445]
Hello all,

First of all I would like to thank you for the awesome initiative (JASP). I use it regularly not only to teach but also to do the great majority of my statistical analyses. It easy to install, it has Windows, MacOs and Linux versions, it is easy to use and easy to report the results. Really amazing software!

I would like to suggest a new feature: the Cochran-Mantel-Haenszel test.

It is used to compare proportions for nominal matched-pairs responses (Agresti).

It compares repeated measures (Before-After) categorical variables that contain more than two levels. For two levels it simplifies to the McNemar test (the request #1014).

I believe it could be placed in Frequencies>Contingency Tables menu below the Vovk-Sellke maximum p-ratio statistic, along side with the future McNemar test.

The report could be just the usual table with the values of the statistics, df and p-values.

Sincerely,

Sal Ramos

**Comment by EJWagenmakers** [comment]
Indeed! @FBartos, can you add this to "more love for contingency tables"?

**Comment by TarandeepKang** [comment]
Hi Team,

I would just like to add some new literature to this discussion.

I have recently seen a new framework for the analysis of proportions:
Laurencelle, L., & Cousineau, D. (2023). Analysis of proportions using arcsine transform with any experimental design. Frontiers in Psychology, 13. https://www.frontiersin.org/articles/10.3389/fpsyg.2022.1045436



And also this suggesting a superior version of the McNemar:

(But my French is to bad to understand much of this)!

Laurencelle, L. (2023). Pour la différence entre deux proportions jumelées, un nouveau test, plus valide et plus puissant [A new standard normal-based test for the difference between paired proportions to supersede obsolete McNemar-like and other indirect procedures]. The Quantitative Methods for Psychology, 19(3), 254–264. https://doi.org/10.20982/tqmp.19.3.p254

#2322 should probably also be considered part of the discussion, but no one has attended to it yet.

**Comment by aboyle57** [comment]
Briefly adding voice to the calls for McNemar's test to be included in JASP! Thanks for all the efforts, it's a great programme.

**Comment by EJWagenmakers** [comment]
Frantisek please oh please :-)

**Comment by TarandeepKang** [comment]
Hi EJ, I don't suppose anybody on your team reads French? In particular in regard to the 2023 paper I posted above, which suggests an improved alternative to the McNemar which is more accurate and powerful indeed the author goes so far in the English version of the title to call it "obsolete". My French is too bad to know if that's grandstanding, but it does sound relevant if you are considering adding the McNemar test?

**Comment by tomtomme** [comment]
@FBartos 
Another voice for the McNemar-Test was raised here: https://github.com/jasp-stats/jasp-issues/issues/3100
It is a standard one in many textbooks. Can we maybe squeeze it in for 0.20?

**Comment by FBartos** [comment]
It is a good time to consolidate some of these contingency table requests and updates. I will see what I can do!

**Comment by tomtomme** [comment]
McNemar is now available in the beta version of the frequencies module, available in the module library.

Cochrans test is also known as Cochrans Q test - changing title accordingly

**Comment by mgarciagranero-debug** [comment]
Hi

Just a comment: Cochran's Q test can be considered as a particular case of Friedman's test when dependent variables are coded as 0/1 (https://www.ncss.com/wp-content/themes/ncss/pdf/Procedures/NCSS/Cochrans_Q_Test.pdf). Therefore, you can use JASP built-in Friedman test to get Cochran's test. Maybe the best position for the test would be under Repeated Measures ANOVA -> Non parametric, and tick a box that states "outcome variables are binary 0/1" to replace the title "Friedman" by "Cochran"

**Comment by Wouterwolf** [comment]
Great that the standard 2x2 McNemar is now added. Is it also possible to do a McNemar-Bowker test, that scales up beyond a 2x2 table to for example a 3x3 table?
